src/storage/btree_load.c

```diff
@@ -4507,40 +4507,40 @@ xbtree_load_online_index (THREAD_ENTRY * thread_p, BTID * btid, const char *bt_n
 	}
     }
 
-  if (pred_stream && pred_stream_size > 0)
+  /* Demote the locks for classes on which we want to load indices. */
+  for (cur_class = 0; cur_class < n_classes; cur_class++)
     {
-      if (stx_map_stream_to_filter_pred (thread_p, &filter_pred, pred_stream, pred_stream_size) != NO_ERROR)
+      ret = lock_demote_class_lock (thread_p, &class_oids[cur_class], new_lock, &old_lock);
+      if (ret != NO_ERROR)
 	{
 	  goto error;
 	}
     }
 
-  if (func_pred_stream && func_pred_stream_size > 0)
+  for (cur_class = 0; cur_class < n_classes; cur_class++)
     {
-      func_index_info.expr_stream = func_pred_stream;
-      func_index_info.expr_stream_size = func_pred_stream_size;
-      func_index_info.col_id = func_col_id;
-      func_index_info.attr_index_start = func_attr_index_start;
-      func_index_info.expr = NULL;
-      if (stx_map_stream_to_func_pred (thread_p, &func_index_info.expr, func_pred_stream,
-				       func_pred_stream_size, &func_unpack_info))
+      if (pred_stream && pred_stream_size > 0)
```

Comment:
please add comments to describe reasons why we re-init filter and function prod for each table.